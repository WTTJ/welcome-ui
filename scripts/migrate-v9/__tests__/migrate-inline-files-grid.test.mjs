import { readFileSync } from 'fs'
import { resolve } from 'path'

import { describe, expect, it } from 'vitest'

import { setupMigrationTest } from '../helpers/test-setup.mjs'
import { migrateAll } from '../index.mjs'

describe('Inline Migration - Grid Components', () => {
  const tempDir = setupMigrationTest('inline-grid-test', '__fixtures__')

  it('migrates Grid components with layout props to div with classNames', async () => {
    // Run migration
    await migrateAll(tempDir, { copyDir: false, interactive: false })

    // Read results
    const migratedComponent = readFileSync(resolve(tempDir, 'GridComponent.tsx'), 'utf8')

    // Verify with snapshots
    expect(migratedComponent).toMatchSnapshot('grid-component.tsx')
  })
})
