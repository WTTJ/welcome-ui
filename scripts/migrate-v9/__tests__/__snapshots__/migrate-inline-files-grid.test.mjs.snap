// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Inline Migration - Grid Components > migrates Grid components with layout props to div with classNames > grid-component.tsx 1`] = `
"import { Grid } from 'welcome-ui/Grid'

export const GridComponent = () => {
  return (
    <div>
      <Grid columns={2} gutter="sm">
        <div>Grid item 1</div>
        <div>Grid item 2</div>
      </Grid>
    </div>
  )
}
"
`;
